pkg/controller/logstorage/dashboards/dashboards_controller.go

```diff
@@ -0,0 +1,376 @@
+		return reconcile.Result{}, err
+	}
+
+	// Get LogStorage resource.
+	logStorage := &operatorv1.LogStorage{}
+	key := utils.DefaultTSEEInstanceKey
+	err = d.client.Get(ctx, key, logStorage)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			d.status.OnCRNotFound()
+			return reconcile.Result{}, nil
+		}
+		d.status.SetDegraded(operatorv1.ResourceReadError, "An error occurred while querying LogStorage", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// Determine where to access Kibana.
+	kibanaHost := "tigera-secure-kb-http.tigera-kibana.svc"
+	kibanaPort := "5601"
+	kibanaScheme := "https"
+
+	var externalKibanaSecret *corev1.Secret
+	if !d.elasticExternal {
+		// Wait for Elasticsearch to be installed and available.
+		elasticsearch, err := utils.GetElasticsearch(ctx, d.client)
+		if err != nil {
+			d.status.SetDegraded(operatorv1.ResourceReadError, "An error occurred trying to retrieve Elasticsearch", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+		if elasticsearch == nil || elasticsearch.Status.Phase != esv1.ElasticsearchReadyPhase {
+			d.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for Elasticsearch cluster to be operational", nil, reqLogger)
+			return reconcile.Result{RequeueAfter: utils.StandardRetry}, nil
+		}
+	} else {
+		// If we're using an external ES and Kibana, the Tenant resource must specify the Kibana endpoint.
+		if tenant == nil || tenant.Spec.Elastic == nil || tenant.Spec.Elastic.KibanaURL == "" {
+			reqLogger.Error(nil, "Kibana URL must be specified for this tenant")
+			d.status.SetDegraded(operatorv1.ResourceValidationError, "Kibana URL must be specified for this tenant", nil, reqLogger)
+			return reconcile.Result{}, nil
+		}
+
+		// Determine the host and port from the URL.
+		url, err := url.Parse(tenant.Spec.Elastic.KibanaURL)
+		if err != nil {
+			reqLogger.Error(err, "Kibana URL is invalid")
+			d.status.SetDegraded(operatorv1.ResourceValidationError, "Kibana URL is invalid", err, reqLogger)
+			return reconcile.Result{}, nil
+		}
+		kibanaScheme = url.Scheme
+		kibanaHost = url.Hostname()
+		kibanaPort = url.Port()
+
+		if tenant.ElasticMTLS() {
+			// If mTLS is enabled, get the secret containing the CA and client certificate.
+			externalKibanaSecret = &corev1.Secret{}
+			err = d.client.Get(ctx, client.ObjectKey{Name: logstorage.ExternalCertsSecret, Namespace: common.OperatorNamespace()}, externalKibanaSecret)
+			if err != nil {
+				reqLogger.Error(err, "Failed to read external Kibana client certificate secret")
+				d.status.SetDegraded(operatorv1.ResourceReadError, "Waiting for external Kibana client certificate secret to be available", err, reqLogger)
+				return reconcile.Result{}, err
+			}
+		}
+	}
+
+	// Query the username and password this Dashboards Installer instance should use to authenticate with Elasticsearch.
+	// ALINA : To check the statement below
```

Comment:
ALINA comment left over :) 

Your comments below are correct for Linseed, but we don't have any logic in the operator to create a user for the dashboard installer job. I think that means we either need to add logic to the operator to create such a secret, or piggy-back on Linseed's.